databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java

```diff
@@ -106,6 +111,41 @@ public static String randRouteHint() {
         return sb.toString();
     }
 
+    private static final char SPECIAL_CHAR = '#';
+
+    public static String uriRouteHint(String URI) {
+        // Encode the URI using Base64
+        String encodedUri = Base64.getEncoder().encodeToString(URI.getBytes());
```

Comment:
a xxhash might be possible here, might allow making this route hint shorter.

(not needed in this pr, could pick it up if one day some people complains it's too long qwq)